python/ray/dag/dag_node.py

```diff
@@ -384,6 +384,20 @@ def __call__(self, node: "DAGNode"):
             if self._stable_uuid in fn.cache:
                 return fn.cache[self._stable_uuid]
 
+        if any(
+            hasattr(child, "get_other_args_to_resolve")
+            and "bind_index" in child.get_other_args_to_resolve()
+            for child in self._bound_args
+        ):
+            self._bound_args = sorted(
+                self._bound_args,
+                key=lambda child: child.get_other_args_to_resolve().get(
+                    "bind_index", float("inf")
+                )
+                if hasattr(child, "get_other_args_to_resolve")
+                else float("inf"),
```

Comment:
This would reorder all bound args. I think the order of bound args should not be changed. For example,

```python
def test_keep_original_order(shared_ray_instance):
    @ray.remote
    class Actor:
        def foo(self, input_data):
            return input_data * 2

        def bar(self, input_data):
            return input_data / 2

        def baz(self, *inputs):
            print(inputs)
            return sum(inputs)

    a = Actor.remote()

    with InputNode() as inp:
        x = a.foo.bind(inp)
        y = a.bar.bind(inp)
        z = a.baz.bind(1, 2, 3, x, y, inp)

        dag = MultiOutputNode([y, x, z])

    ray.get(dag.execute(4))
```

`z` should print 1, 2, 3, x, y, inp. But the args are reordered as: x, y, 1, 2, 3, inp.